CMakeLists.txt

```diff
@@ -28,6 +28,15 @@ option(use_cppunittest "set use_cppunittest to ON to build CppUnitTest tests on
 option(run_traceability "run traceability tool (default is ON)" ON)
 option(run_reals_check "set run_reals_check to ON to run reals check (default is OFF)." OFF)
 
+if((NOT "${CMAKE_BUILD_TYPE}" STREQUAL "RelWithDebInfo") AND (NOT "${CMAKE_BUILD_TYPE}" STREQUAL "Debug"))
+    message(FATAL_ERROR "CMAKE_BUILD_TYPE must be specified and be precisely one of Debug or RelWithDebInfo")
+endif()
+
+set(CMAKE_CONFIGURATION_TYPES "${CMAKE_BUILD_TYPE}" CACHE STRING "" FORCE)

```

Comment:
>CMAKE_CONFIGURATION_TYPES [](http://example.com/codeflow?start=4&length=25)

a comment explaining why this line is needed would be really nice for future maintainers of the code.

Reply:
Added more comments